src/vs/workbench/parts/terminal/electron-browser/terminalProcess.js

```diff
@@ -48,6 +48,7 @@ ptyProcess.on('data', function (data) {
 });
 
 ptyProcess.on('exit', function (exitCode) {
+	ptyProcess.kill();
```

Comment:
Tested this in Windows too.

So when I start Integrated terminal in VS Code, I see one extra instance of bash (in Linux or PowerShell in Windows) and when I exit VS Code, that instance disappears from the task manager. Let me know if there's anything else you think I should be testing for this.